synapse/storage/databases/main/account_data.py

```diff
@@ -353,12 +384,15 @@ async def ignored_by(self, user_id: str) -> Set[str]:
             )
         )
 
-    def process_replication_rows(self, stream_name, instance_name, token, rows):
+    def process_replication_rows(
+        self,
+        stream_name: str,
+        instance_name: str,
+        token: int,
+        rows: Iterable[Any],
+    ) -> None:
         if stream_name == TagAccountDataStream.NAME:
             self._account_data_id_gen.advance(instance_name, token)
-            for row in rows:
-                self.get_tags_for_user.invalidate((row.user_id,))
-                self._account_data_stream_cache.entity_has_changed(row.user_id, token)
```

Comment:
At some point in the past, `TagsWorkerStore` was split out from `AccountDataWorkerStore` (ca9b9d970374766f7b53dc5b1762cfa1b0e1a5dd). I think this was missed.

I left the call to `advance` in, in case we ever instantiate an `AccountDataWorkerStore` that is not a `TagsWorkerStore`.
`advance` ought to be idempotent so it should be safe to call it twice.

I think we don't need to call `entity_has_changed` here, since it only impacts one spot, which doesn't appear to have anything to do with tags:
https://github.com/matrix-org/synapse/blob/0cc3bf97b4399234cf20f52ae4c09d03661225ff/synapse/storage/databases/main/account_data.py#L326

Reply:
> I left the call to advance in, in case we ever instantiate an AccountDataWorkerStore that is not a TagsWorkerStore.

Say we did instantiate such a store. What does leaving the `advance` call in gain us? It feels a bit odd to have something that looks like it's responding/handling to tag changes, but not really do anything with it.

Reply:
`_account_data_id_gen.get_current_token()` would lag behind since it's the minimum stream position across all writers. In the worst case it'd get stuck entirely if there is a writer that only writes tag data.

I think the effect of that would be that various bits of Synapse would not get notified about data written after `get_current_token()`, since it can't be sure that all writes had completed.